labs/lab6/lab6.adoc

```diff
@@ -29,56 +37,311 @@ Password: INSTRUCTOR WILL PROVIDE
+
+Returning to the Prometheus rule again, notice how this rule wil never fire as it is impossible for a count such as this to have a negative number. This was done intentionally during the lab preparation to ensure an alert did not accidentally fire. 
+
+In this lab, the minimum number of application nodes will be modified from 0 to 2, or 1 more than are currently running.
+
+The configuration for Prometheus and AlertManager are defined in ConfigMaps and mounted as volumes into each container. Using the OpenShift CLI, first switch into the `openshift-metrics` project.
+
+.master$
+[source, bash]
+----
+oc project openshift-metrics
+----
+
+Now, list all defined ConfigMaps
+
+.master$
+[source, bash]
+----
+oc get configmaps
+----
+
+To modify the Prometheus rules, edit the prometheus ConfigMap
+
+.master$
+[source, bash]
+----
+oc edit configmap prometheus
+----
+
+Locate the rule that is defined in the key `prometheus.additional.rules`
+
+Modify the expression for the _Minimum Number of Application Nodes_ rule as shown below:
+
+[source, bash]
+----
+sum(up{type="app",job="kubernetes-nodes"}) < 2
+----
+
+One of the benefits of ConfigMaps is that once a change has been made, the resulting value is made available in the pod. 
+
+With the value of the rule file changed, Prometheus can be instructed to reload its own configuration based on the updated values without having to stop the instance or the pod. This can be done by making a request against the API. The easiest method to do so is to obtain a remote session inside the pod.
+
+Execute the following command to access the Prometheus container:
+
+.master$
+[source, bash]
+----
+oc rsh -c prometheus $(oc get pods -l=app=prometheus -o 'jsonpath={.items[0].metadata.name}')
+----
+
+Once inside the container, confirm the file mounted in the ConfigMap has been updated.
+
+.prometheus container$
+[source, bash]
+----
+cat /etc/prometheus/prometheus.additional.rules
+----
+
+It may take up to 1 minute for the value to change. Continue executing the prior command until a change is detected.
+
+Once the value in the files has changed, to force Prometheus to reload its configuration, execute the following request
+
+.prometheus container$
+[source, bash]
+----
+curl -X POST http://localhost:9090/-/reload
```

Comment:
should the exit they container after this?

Reply:
Added explaination